drivers/ruby/lib/net.rb

```diff
@@ -677,20 +677,40 @@ def init_socket
         context = create_context(@ssl_opts)
         @socket = OpenSSL::SSL::SSLSocket.new(@tcp_socket, context)
         @socket.sync_close = true
-        @socket.connect
+        socket_connect
         verify_cert!(@socket, context)
       else
         @socket = base_socket
+        socket_connect
       end
     end
 
     def base_socket
-      socket = TCPSocket.open(@host, @port)
+      @socket.close if @socket
+      socket = Socket.new(Socket::AF_INET, Socket::SOCK_STREAM, 0)
       socket.setsockopt(Socket::IPPROTO_TCP, Socket::TCP_NODELAY, 1)
       socket.setsockopt(Socket::SOL_SOCKET, Socket::SO_KEEPALIVE, 1)
       socket
     end
 
+    def socket_connect 
```

Comment:
What's the reason why this is called by `init_socket` rather than `base_socket`?  Is it not correct to finish connecting the socket before `base_socket` returns?